### PR TITLE
feat(tools): add short description, rework metadata management

### PR DESCRIPTION
**Pull request by kapetr**
Refs: https://github.ibm.com/Incubation/bee-ui/issues/636

### CODE REVIEW

File: src/app/auth/index.ts [hunk 1]
```diff
@@ -120,11 +123,15 @@ const authResult = NextAuth(() => ({
...
       try {
-        user.apiUser = await createUser(account.access_token, {
+        const result = await createUser(account.access_token, {
           metadata: encodeMetadata<UserMetadata>({
             email: user.email ?? '',
           }),
         });
+        if (result) {
+          user.userEntity = user.userEntity =
+            decodeEntityWithMetadata<UserEntity>(result);
```
Comment (PetrBulanek) [inline src/app/auth/index.ts:133]: Is this on purpose?

File: src/modules/chat/assistant-plan/PlanStep.tsx [hunk 2]
```diff
@@ -264,9 +262,9 @@ export function PlanStep({ step, toolCall }: Props) {
                   </div>
                 </div>
               ) : (
-                <AnimatePresence>
```
Comment (PetrBulanek) [inline src/modules/chat/assistant-plan/PlanStep.tsx:267]: Does it work without the `AnimatePresence`?

**Review by PetrBulanek — APPROVED** [review]

File: src/app/auth/index.ts (hunk 1 above)
Comment (kapetr) [reply src/app/auth/index.ts:133]: No, thanks!

File: src/modules/chat/assistant-plan/PlanStep.tsx (hunk 2 above)
Comment (kapetr) [reply src/modules/chat/assistant-plan/PlanStep.tsx:267]: I was just testing something to get rid of 'unique "key" prop' errors in MotionComponent (I didn't resolve it). As a matter of fact it should only be required to enable exit animations according to the documentation, but it seems that it's not working without it, reverting.

**Pull request merged**